[MTM-64398]: Add information about the newly added CA feature to the section which covers integration using REST API (#3618)
SoftwareAG/c8y-docs: content/device-integration/device-integration-rest-bundle/device-authentication.md

Mutual TLS (mTLS) is a security protocol that uses X.509 certificates for both client and server authentication in a communication session.

> The mTLS protocol is commonly used to secure conn[data-broker](../../data-broker)ections in web services, APIs, and other networked applications. When generating tokens using mTLS, the process involves the authentication of both the client and the server using X.509 certificates.

Reviewer: ```suggestion
The mTLS protocol is commonly used to secure [connections](../../data-broker) in web services, APIs, and other networked applications. When generating tokens using mTLS, the process involves the authentication of both the client and the server using X.509 certificates.
```

Retrieving device access tokens from the platform with certificates does not require the tenant ID, username and password. Authentication information will be obtained from the certificates.
The device access token can be retrieved by sending only the device leaf certificate if an immediate issuer of the device certificate is uploaded to the trusted certificates list. If the uploaded trusted certificate is not an immediate issuer of the device certificate but belongs to the device's chain of trust, then the device must send the entire certificate chain in the `X-Ssl-Cert-Chain` to be authenticated successfully and retrieve the device access token.

> Trusted certificates (root/intermediate) can be uploaded to the {{< product-c8y-iot >}} via an endpoint. For details, see [Trusted certificates](/device-certificate-authentication/device-certificates).

Reviewer 2: The doc should make it clear that customers have 2 options to provision the trusted certificate. The customer cam either upload the certificate (as before) or can use the CA feature for that. 
Proposal:
```suggestion
You can define which organization you trust by uploading the CA certificate (root/intermediate) to {{< product-c8y-iot >}}. For details, see [Trusted certificates](/device-certificate-authentication/device-certificates). Alternatively, you can use the Certificate Authority feature. In this case, the root CA certificate is created by {{< product-c8y-iot >}}, along with a device certificate that is signed by this CA. For details, see [Certificate authority](/device-certificate-authentication/certificate-authority).
```

> Trusted certificates (root/intermediate) can be uploaded to the {{< product-c8y-iot >}} via endpoint, for more information please visit [Trusted Certificates](/device-certificate-authentication/device-certificates).

Reviewer: ```suggestion
Trusted certificates (root/intermediate) can be uploaded to the {{< product-c8y-iot >}} via an endpoint. For details, see [Trusted certificates](/device-certificate-authentication/device-certificates).
```

> The Certificate Authority (root) certificate is created for each tenant. Moreover, a device certificate, which is signed by the CA, is created. For details, see [Certificate authority](/device-certificate-authentication/certificate-authority).

Reviewer 2: Included in the proposal above
```suggestion
```

> Certificate Authority (root) certificate will be created for each tenant and device certificate also created which is signed by CA, for more information please visit [Certificate Authority](/device-certificate-authentication/certificate-authority).

Reviewer: ```suggestion
The Certificate Authority (root) certificate is created for each tenant. Moreover, a device certificate, which is signed by the CA, is created. For details, see [Certificate authority](/device-certificate-authentication/certificate-authority).
```
